src/annotation.js

```diff
@@ -0,0 +1,607 @@
+var $ = require('jquery');
+var inherit = require('./inherit');
+var geo_event = require('./event');
+var transform = require('./transform');
+
+var annotationId = 0;
+
+var annotationState = {
+  create: 'create',
+  done: 'done',
+  edit: 'edit'
+};
+
+/////////////////////////////////////////////////////////////////////////////
+/**
+ * Base annotation class
+ *
+ * @class geo.annotation
+ * @param {string} type the type of annotation.
+ * @param {object?} options Inidividual annotations have additional options.
+ * @param {string} [options.name] A name for the annotation.  This defaults to
+ *    the type with a unique ID suffixed to it.
+ * @param {geo.annotationLayer} [options.layer] a reference to the controlling
+ *    layer.  This is used for coordinate transforms.
+ * @param {string} [options.state] initial annotation state.  One of the
+ *    annotation.state values.
+ * @returns {geo.annotation}
+ */
+/////////////////////////////////////////////////////////////////////////////
+var annotation = function (type, args) {
```

Comment:
At the moment there isn't an annotation registry.  I think it would be good to have one, as this (a) allows easily enumerating available annotations, and (b) each annotation can list its required features, so when you ask for an annotation layer, instead of asking for a specific renderer or a list of required features, you could alternately supply a list of annotations that would be used and that would be used to select an appropriate renderer.

It isn't much work to add that.  I can do it now or deferred it to the next PR.